bot/utility/utility.py

```diff
@@ -206,6 +208,48 @@ async def welcome_message(self, user: discord.Member):
                                                               "__gemeinsam__ schaffen wir das!** :muscle:")
         await user.send(content=content, embed=embed)
 
+    @commands.command(name='format')
+    async def format(self, ctx: commands.Context, language: str, *code: str):
+        """Handler for the `format` command. Formats the provided message using clang-format and prints it using markdown syntax highlighting
+            Refer to the !howto format for an extended explanation of how to use it
+        Args:
+            language: The language abbreviation to highlight the resulting code with
+
+        """
+        try:
+            code = " ".join(code) if ctx.message.reference is None else ctx.message.reference.resolved.content
+            # mypy would erroneously find an error in the next two lines because it doesn't notice that at this point, `code` is always a str.
+            # You can verify its correctness by renaming the variable `code` as assignment target in the previous and all subsequent lines
+            code = code.removeprefix("```").removeprefix("`").removesuffix("```").removesuffix("`")   # type: ignore # this allows the command to format single- and multiline code blocks
+            result = subprocess.run("clang-format", text=True, input=code.replace("```", "´´´"), check=True, capture_output=True)  # type: ignore
+            embed = discord.Embed(title="Formatted", description=f"```{language}\n{result.stdout}\n```")
+            await ctx.send(embed=embed)
+        except Exception as error:
+            log.error("Failed to format code: %s", error)
+            await self.howto_format(ctx)
```

Comment:
What happens now, when language is not passed?

Reply:
Nothing visible to the discord user happens. Internally, the error handler handles the exception and logs its stack trace.

I would suggest that instead, an error message (something like "Your command missed an argument. Check !howto for the correct syntax") gets shown to the user.
However, as this would change the behaviour of all commands, I consider this to be outside the scope of this pull request.

If **only**  the language gets passed, e.g. `!format c++`, the language is printed. I don't see an issue in that, but if desired, an exception could be thrown to be handled by the error handler.